include/oneapi/tbb/parallel_for_each.h

```diff
@@ -118,14 +118,17 @@ struct feeder_item_task: public task {
     using feeder_type = feeder_impl<Body, Item>;
 
     template <typename ItemType>
-    feeder_item_task(ItemType&& input_item, feeder_type& feeder, small_object_allocator& alloc) :
+    feeder_item_task(ItemType&& input_item, feeder_type& feeder, small_object_allocator& alloc, wait_tree_vertex_interface* node) :
```

Comment:
What do you think about passing the `wait_context_vertex` to the constructor and call `get_thread_reference_vertex` internally?
Current API also allows to pass `wait_context_vertex` into this constructor and associate the task with the wait context itself and the logic why this or another `wait_tree_vertex_interface` was chosen is kept outside of the task. 
If we believe that if we want to make sure that per-thread ref node for each feeder task - it makes sense to move the call into the constructor.

Reply:
Done